### PR TITLE
marketplace stats api added

### PR DESCRIPTION
**Pull request by rex-god**
Stats api added, date filter added. `start_date` & `end_date` holds date in _YYYY-MM-DD_ format.
`end_date` is optional query param and by default value is today.

### CODE REVIEW

File: src/modules/marketplace/open-api.ts [hunk 1]
```diff
@@ -562,3 +562,25 @@ export function OrderHistoryOpenApi() {
...
+		ApiOkResponse({
+			description: 'Marketplace stats fetched successfully',
+			schema: {
+				example: {
+					success: true,
+					message: 'Marketplace stats fetched successfully',
+					result: {
+						sales_count: 1,
+						sales_value: 0.1,
+						no_of_seller: 1,
```
Comment (codinpsycho) [inline src/modules/marketplace/open-api.ts:578]: Should be seller_count to keep naming consistent.

File: src/modules/marketplace/dto/get-stats.dto.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  @ApiProperty({
+		title: 'Marketplace Address',
+		type: String,
+		description: 'address of the Marketplace in which the NFT is being sold',
+		example: '54K5BTMj8ynktzEeaD1iGhAJsCN2svBKmD1fTQTonbBB',
+	})
+	@IsNotEmpty()
+	@IsString()
+	readonly marketplace_address: string;
```
Comment (codinpsycho) [inline src/modules/marketplace/dto/get-stats.dto.ts:24]: This should also be optional, or else there need to be a separate mechanism to fetch all time data.

File: src/modules/marketplace/dto/get-stats.dto.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  @ApiProperty({
+		title: 'start_date',
+		type: Date,
+		description: 'YYYY-MM-DD format',
+		example: '2022-09-02',
+	})
+	@IsNotEmpty()
+  @IsDate()
+  @Transform(({ value }) => new Date(value))
+	readonly start_date: Date;
```
Comment (codinpsycho) [inline src/modules/marketplace/dto/get-stats.dto.ts:35]: This should also be optional, or else there need to be a separate mechanism to fetch all time data.

File: src/modules/marketplace/open-api.ts (hunk 1 above)
Comment (rex-god) [reply src/modules/marketplace/open-api.ts:578]: Ok

File: src/modules/marketplace/dto/get-stats.dto.ts (hunk 3 above)
Comment (codinpsycho) [reply src/modules/marketplace/dto/get-stats.dto.ts:35]: What is the start_date value required to fetch all time data?

File: src/modules/marketplace/listing-service.ts [hunk 4]
```diff
@@ -337,4 +338,13 @@ export class ListingService {
 			throw newProgramErrorFrom(err, "get_purchases_error");
 		}
 	}
+
+	async stats(getStatsDto: GetStatsDto): Promise<any> {
+		try {
+			const result = await this.listingRepo.stats(getStatsDto);
+			return result;
+		} catch (err) {
+			throw new HttpException('No data found!', HttpStatus.NOT_FOUND);
```
Comment (vgvishesh) [inline src/modules/marketplace/listing-service.ts:347]: Change this to newProgramError

File: src/modules/marketplace/open-api.ts [hunk 5]
```diff
@@ -562,3 +562,25 @@ export function OrderHistoryOpenApi() {
...
+	return applyDecorators(
+		ApiOperation({ summary: 'Marketplace stats' }),
+		ApiOkResponse({
+			description: 'Marketplace stats fetched successfully',
+			schema: {
+				example: {
+					success: true,
+					message: 'Marketplace stats fetched successfully',
+					result: {
+						sales_count: 1,
```
Comment (vgvishesh) [inline src/modules/marketplace/open-api.ts:576]: These should be renamed to:
sales_count -> total_sales
sales_value -> sales_volume
seller_count -> total_sellers
listed_count -> total_listings
listed_value -> listed_volume

File: src/modules/marketplace/mp-controller.ts [hunk 6]
```diff
@@ -222,4 +223,17 @@ export class CreateMarketplaceController {
 			result: result,
 		}
 	}
+
+	@Get('stats')
+	@StatsOpenApi()
+	@Version('1')
+	async stats(@Query() getStatsDto: GetStatsDto): Promise<any> {
+		// console.log("get active listings of a seller in market request received");
```
Comment (vgvishesh) [inline src/modules/marketplace/mp-controller.ts:231]: Keep the log with an updated and relevant message. Check other mp apis.

File: src/modules/marketplace/listing-service.ts (hunk 4 above)
Comment (rex-god) [reply src/modules/marketplace/listing-service.ts:347]: No need, cause fetching from DB.

**Review by codinpsycho — APPROVED** [review]

**Pull request merged**